lisa/sut_orchestrator/baremetal/schema.py

```diff
@@ -134,6 +141,20 @@ def __post_init__(self, *args: Any, **kwargs: Any) -> None:
         add_secret(self.password)
 
 
+@dataclass_json()
+@dataclass
+class TftpBuildSchema(BuildSchema):
+    connection: Optional[schema.RemoteNode] = field(
+        default=None, metadata=field_metadata(required=True)
+    )
+    share: str = ""
+    server_name: str = ""
+
+    def __post_init__(self, *args: Any, **kwargs: Any) -> None:
+        if self.connection:
```

Comment:
`if self.connection and self.connection.password:`

The auth may be key based.

Reply:
The change is done, thanks